homeassistant/components/kef/media_player.py

```diff
@@ -392,28 +399,33 @@ def icon(self):
         """Send next track command."""
         await self._speaker.next_track()
 
-    async def update_dsp(self):
+    async def update_dsp(self) -> None:
         """Update the DSP settings."""
+        if self._speaker_type == "LS50" and self._state == STATE_OFF:
+            # The LSX is able to respond when off the LS50 has to be on.
+            return
+
         mode = await self._speaker.get_mode()
-        self._dsp = dict(
-            **mode._asdict(),
-            desk_db=await self._speaker.get_desk_db(),
-            wall_db=await self._speaker.get_wall_db(),
-            treble_db=await self._speaker.get_treble_db(),
-            high_hz=await self._speaker.get_high_hz(),
-            low_hz=await self._speaker.get_low_hz(),
-            sub_db=await self._speaker.get_sub_db(),
+        (mode, *rest) = await asyncio.gather(
+            self._speaker.get_mode(),
+            self._speaker.get_desk_db(),
+            self._speaker.get_wall_db(),
+            self._speaker.get_treble_db(),
+            self._speaker.get_high_hz(),
+            self._speaker.get_low_hz(),
+            self._speaker.get_sub_db(),
         )
-        return self._dsp
+        keys = ["desk_db", "wall_db", "treble_db", "high_hz", "low_hz", "sub_db"]
+        self._dsp = dict(zip(keys, rest), **mode._asdict())
```

Comment:
You seem to keep going through great lengths to avoid writing things out 😆

Reply:
Haha, it comes naturally to me 😂